keras_cv/models/regnet.py

```diff
@@ -0,0 +1,1524 @@
+      group_width: group width of all blocks in  this stage
+      filters_in: input filters to this stage
+      filters_out: output filters from this stage
+      name: name prefix
+
+    Returns:
+      Output tensor of Stage
+    """
+    if name is None:
+        name = str(backend.get_uid("stage"))
+
+    def apply(inputs):
+        x = inputs
+        if block_type == "X":
+            x = XBlock(
+                filters_in,
+                filters_out,
+                group_width,
+                stride=2,
+                name=f"{name}_XBlock_0",
+            )(x)
+            for i in range(1, depth):
+                x = XBlock(
+                    filters_out,
+                    filters_out,
+                    group_width,
+                    name=f"{name}_XBlock_{i}",
+                )(x)
+        elif block_type == "Y":
+            x = YBlock(
+                filters_in,
+                filters_out,
+                group_width,
+                stride=2,
+                name=name + "_YBlock_0",
+            )(x)
+            for i in range(1, depth):
+                x = YBlock(
+                    filters_out,
+                    filters_out,
+                    group_width,
+                    name=f"{name}_YBlock_{i}",
+                )(x)
+        elif block_type == "Z":
+            x = ZBlock(
+                filters_in,
+                filters_out,
+                group_width,
+                stride=2,
+                name=f"{name}_ZBlock_0",
+            )(x)
+            for i in range(1, depth):
+                x = ZBlock(
+                    filters_out,
+                    filters_out,
+                    group_width,
+                    name=f"{name}_ZBlock_{i}",
+                )(x)
+        else:
+            raise NotImplementedError(
+                f"Block type `{block_type}` not recognized."
+                f"block_type must be one of (`X`, `Y`, `Z`). "
+            )
+        return x
+
+    return apply
+
+
+def Head(num_classes=None, name=None, activation=None):
+    """Implementation of classification head of RegNet.
+
+    Args:
+      num_classes: number of classes for Dense layer
+      name: name prefix
+
+    Returns:
+      Output logits tensor.
+    """
+    if name is None:
+        name = str(backend.get_uid("head"))
+
+    def apply(x):
+        x = layers.GlobalAveragePooling2D(name=name + "_head_gap")(x)
+        x = layers.Dense(num_classes, name=name + "head_dense", activation=activation)(
+            x
+        )
+        return x
+
+    return apply
+
+
+def RegNet(
+    depths,
+    widths,
+    group_width,
+    block_type,
+    include_rescaling,
+    include_top,
+    model_name="regnet",
+    weights=None,
+    input_shape=(None, None, 3),
+    pooling=None,
+    num_classes=None,
+    classifier_activation="softmax",
+    **kwargs,
+):
+    """Instantiates RegNet architecture given specific configuration.
```

Comment:
While this docstring is on an internal method, whatever changes are made to the BASE_DOCSTRING above should also be made here for consistency.